src/commands_controller.ts

```diff
@@ -107,7 +107,15 @@ export class CommandsController implements Disposable, NeovimExtensionRequestPro
 
     /// SCROLL COMMANDS ///
     private scrollPage = (by: "page" | "halfPage", to: "up" | "down"): void => {
-        vscode.commands.executeCommand("editorScroll", { to, by, revealCursor: true });
+        const editor = vscode.window.activeTextEditor;
+        if (!editor) {
+            return;
+        }
+        const visibleRanges = editor.visibleRanges;
+        const scrollLines = Math.ceil(
+            (visibleRanges[0].end.line - visibleRanges[0].start.line) / (by === "halfPage" ? 2 : 1),
```

Comment:
I think you should take both first and last visible ranges, since for folds there can be few entries here

Reply:
You can use this function:
https://github.com/asvetliakov/vscode-neovim/blob/266817e84d9e0f6bf63579f5c0798844b47fdf5f/src/utils.ts#L534-L543

and modify it to return `Range | undefined` rather than boolean